### PR TITLE
project-schema: Add Organization.beneficialOwners

### LINKED ISSUE #399: beneficialOwners: incorporate beneficialOwners information using OCDS extension
# Background

This issue relates to the following CoST IDS elements proposed in the CoST IDS/OC4IDS review:

<details>

<summary> Beneficial ownership </summary>

## Beneficial ownership

**Module:** Institutional
**Indicator:** Access to Information

### Disclosure format

> ​​D​isclose the ​beneficial owners​ of the contractors and suppliers appointed in the project (E.g. [​name, identifier​]).  

### OC4IDS mapping

> Use modelling from[ OCDS Beneficial Owners extension](https://extensions.open-contracting.org/en/extensions/beneficialOwners/master/) 

</details>

# Proposal

Add a `beneficialOwners` object to the `parties` array following the modelling from the [OCDS Beneficial Owners](https://extensions.open-contracting.org/en/extensions/beneficialOwners/master/) extension.

# Example

```json
{
  "parties": [
    {
      "id": "AHL",
      "name": "Alpha Holdings Ltd",
      "beneficialOwners": [
        {
          "id": "1",
          "name": "Juan Perez",
          "identifier": {
            "scheme": "PRY-IDCARD",
            "id": "12345"
          },
          "nationalities": [
            "PY"
          ],
          "address": {
            "streetAddress": "Avenida Eusebio Ayala 1347",
            "locality": "Asunción",
            "region": "Gran Asunción",
            "postalCode": "1001",
            "countryName": "Paraguay"
          },
          "email": "jperez@example.com",
          "faxNumber": "+595210000001",
          "telephone": "+595210000000"
        }
      ]
    }
  ]
}

```

cc @evelyndinora

### PR DESCRIPTION
**Pull request by duncandewhurst**
**Related issues**

* Closes #399

**Description**

`manage.py` updates:
* Add beneficialOwners extension to profile
* Update 'paths to skip' logic in `update_sub_schema_reference`

**Merge checklist**

- [x] [Log your changes](https://ocds-standard-development-handbook.readthedocs.io/en/latest/standard/contributing.html#logging-changes)

If there are changes to `project-schema.json` or `project-package-schema.json`:

- [x] Update the examples:
  - [x] `docs/examples/example.json`
  - [x] `docs/examples/blank.json`
- [x] Run `./manage.py pre-commit` to update `docs/_static/i8n.csv`

If you added a new definition to the schema, run `./manage.py pre-commit`.

If you added a new codelist:

  - [x] Add an entry to `docs/reference/codelists.md`

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -10,4 +10,4 @@ jobs:
         with:
           python-version: '3.10'
       - run: pip install codespell
-      - run: codespell -S .git,docson,locale,currency.csv -L tne,zar .
+      - run: codespell -S .git,docson,locale,currency.csv -L tne,zar,fo,nam .
```
Comment (jpmckinney) [inline .github/workflows/spellcheck.yml]: We can exclude external codelists (country.csv has "Viet Nam").
```suggestion
      - run: codespell -S .git,docson,locale,country.csv,currency.csv -L fo,tne,zar .
```

File: docs/reference/changelog.md [hunk 2]
```diff
@@ -38,6 +39,7 @@
   * milestoneType
   * milestoneStatus
   * milestoneCode
+* [#427](https://github.com/open-contracting/infrastructure/pull/427) - add country codelist.
```
Comment (jpmckinney) [inline docs/reference/changelog.md]: Would be clearer to format as:

```md
* add codelists:
  * [#369](https://github.com/open-contracting/infrastructure/pull/369) - classificationScheme
  * [#427](https://github.com/open-contracting/infrastructure/pull/427) - country
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneType
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneStatus
  * [#426](https://github.com/open-contracting/infrastructure/pull/426) - milestoneCode
```

File: manage.py [hunk 3]
```diff
@@ -334,21 +334,32 @@ def update_sub_schema_reference(schema):
...
+                         'parties/0/people/0/address',
+                         'parties/0/people/0/identifier']
 
         # Add examples
         definition["references"] = get_definition_references(schema, defn)
         for ref in definition["references"]:
-            if ref[0] not in schema['definitions'] and '/'.join(ref) not in paths_to_skip:
-                if ref[-1] == '0':
-                    ref.pop(-1)
+            if ref[0] not in schema['definitions']:
```
Comment (jpmckinney) [inline manage.py]: The logic is easier to read as:

```python
            if ref[0] not in schema['definitions'] and not any(p == '/'.join(ref)[:len(p)] for p in paths_to_skip):

**Review by jpmckinney — APPROVED** [review]
Some easy to resolve comments. Feel free to merge once resolved.

File: docs/reference/changelog.md (hunk 2 above)
Comment (duncandewhurst) [reply docs/reference/changelog.md]: Done in 29cb3e0cce565b00e0e4e2ab82178bc68a2bbebe

File: manage.py (hunk 3 above)
Comment (duncandewhurst) [reply manage.py]: Ah, that's much nicer! Done in 8e0209c5b89af57b62ab60127837b7f68399e3f9

**Pull request merged**